golem-rib/src/type_inference/mod.rs

```diff
@@ -956,17 +977,13 @@ mod type_inference_tests {
             assert_eq!(expr, expected);
         }
 
-        // TODO; none is un-inferred, probably due to unable-to-do-anything
-        // for pull up and push down phases
         #[test]
         fn test_pattern_match_with_result() {
             let rib_expr = r#"
               let x: u64 = 1;
 
-              let fallback: option<option<u64>> = none;
-
               match err(x) {
-                err(_) => fallback,
+                err(_) => none,
                 ok(_) => some(some(x))
               }
             "#;
```

Comment:
Test that shows better type inference. With multiple passes of pushing down and propagating types `none` inferred to be `Option(Option(str))`